src/lib.rs

```diff
@@ -139,23 +139,26 @@ impl<P: Key> BeforeMiddleware for Write<P> where P::Value: Send {
 }
 
 impl<P: Key> AfterMiddleware for State<P> where P::Value: Send + Sync {
-    fn after(&self, _: &mut Request, res: &mut Response) -> IronResult<()> {
-        res.extensions.insert::<State<P>>(self.data.clone());
-        Ok(())
+    fn after(&self, _: &mut Request, res: Response) -> IronResult<Response> {
+        let mut outres = res;
```

Comment:
That doesn't build:

```
src/lib.rs:142:5: 145:6 error: method `after` has an incompatible type for trait: expected struct `iron::response::Response`, found &-ptr [E0053]
src/lib.rs:142     fn after(&self, _: &mut Request, mut res: &mut Response) -> IronResult<Response> {
src/lib.rs:143         res.extensions.insert::<State<P>>(self.data.clone());
src/lib.rs:144         Ok(res)
src/lib.rs:145     }
src/lib.rs:144:12: 144:15 error: mismatched types:
 expected `iron::response::Response`,
    found `&mut iron::response::Response`
(expected struct `iron::response::Response`,
    found &-ptr) [E0308]
src/lib.rs:144         Ok(res)
                          ^~~
error: aborting due to 2 previous errors
Could not compile `persistent`.
```

Did you intend `Response` to be a mutable borrow in the [trait definition](https://github.com/iron/iron/blob/6c15f214869815dde9bf7540a6fc18d966516c41/src/middleware/mod.rs#L95)?